src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2398,11 +2398,16 @@ bool SessionImpl::deleteTorrent(const TorrentID &id, const DeleteOption deleteOp
 
 bool SessionImpl::cancelDownloadMetadata(const TorrentID &id)
 {
-    const auto downloadedMetadataIter = m_downloadedMetadata.find(id);
-    if (downloadedMetadataIter == m_downloadedMetadata.end())
+    const auto downloadedMetadataIter = asConst(m_downloadedMetadata).find(id);
```

Comment:
why not
```suggestion
    const auto downloadedMetadataIter = m_downloadedMetadata.constFind(id);
```

Reply:
To keep it STL compatible?.. 
Although I don't give a damn about it, to be honest.

I just wanted to get rid of the clang code analyzer warning about mixing iterators. But now it issues another warning that cannot be eliminated. So maybe just roll back this change?

Reply:
> > To keep it STL compatible?..
> 
> Not at all. It is just the usual method for getting a const iterator.

You didn't understand. I mean that `constFind()` isn't STL compatible.

Reply:
>I mean that constFind() isn't STL compatible.

I don't care about that. Were there any concern about it?

Reply:
> > But now it issues another warning that cannot be eliminated. So maybe just roll back this change?
> 
> It is hard to discuss without any info.

It insists that I change `const lt::torrent_handle nativeHandle = downloadedMetadataIter.value();` to use const reference, i.e. `const lt::torrent_handle &nativeHandle = downloadedMetadataIter.value();`. But it does not take into account that it will become dangling at the next line:
```c++
m_downloadedMetadata.erase(downloadedMetadataIter);
```

Reply:
In any case, this is not a problem of this PR, so I rolled back these changes.

Reply:
> I don't give a damn about it, to be honest.

As I said above:
>I don't give a damn about it, to be honest.

I don't know what prompted me to go this way...